### ISSUE TITLE
"The output model HDF5 file was not found" error from SymmetrizeCalculation

### ISSUE THREAD

**Issue by minkyu-p** [greschd/aiida-tbextraction#23]
Dear all,

I faced a problem while I run the example script included in aiida-tbextraction. 
The error message is following.

From `verdi process show 1560`
```
Property     Value
-----------  --------------------------------------------------------
type         SymmetrizeCalculation
state        Finished [300] The output model HDF5 file was not found.
pk           1560
uuid         20435e24-5d32-4c40-a3db-c9894551427c
label
description
ctime        2021-10-01 01:24:22.998931+00:00
mtime        2021-10-01 01:24:55.780517+00:00
computer     [4] localhost

Inputs        PK  Type
----------  ----  --------------
code        1145  Code
symmetries  1431  SinglefileData
tb_model    1557  SinglefileData

Outputs          PK  Type
-------------  ----  ----------
remote_folder  1562  RemoteData
retrieved      1582  FolderData

Caller      PK  Type
--------  ----  -----------------------
CALL      1509  TightBindingCalculation

Log messages
---------------------------------------------
There are 1 log messages for this calculation
Run 'verdi process report 1560' to see them
```

From `verdi process report 1560`

```
*** 1560: None
*** (empty scheduler output file)
*** Scheduler errors:
Traceback (most recent call last):
  File "/home/user3/minkyu/.local/bin/tbmodels", line 8, in <module>
    sys.exit(cli())
  File "/home/user3/minkyu/anaconda3/lib/python3.8/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/home/user3/minkyu/anaconda3/lib/python3.8/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/home/user3/minkyu/anaconda3/lib/python3.8/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/user3/minkyu/anaconda3/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/user3/minkyu/anaconda3/lib/python3.8/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/home/user3/minkyu/.local/lib/python3.8/site-packages/tbmodels/_cli.py", line 161, in symmetrize
    sym = sr.io.load(symmetries)
  File "/home/user3/minkyu/.local/lib/python3.8/site-packages/symmetry_representation/io.py", line 30, in load
    return fsc.hdf5_io.load(hdf5_file)
  File "/home/user3/minkyu/.local/lib/python3.8/site-packages/fsc/hdf5_io/_save_load.py", line 157, in from_hdf5_file
    return from_hdf5(f)
  File "/home/user3/minkyu/.local/lib/python3.8/site-packages/fsc/hdf5_io/_save_load.py", line 63, in from_hdf5
    type_tag = hdf5_handle[TYPE_TAG_KEY][()].decode('utf-8')
AttributeError: 'str' object has no attribute 'decode'

*** 1 LOG MESSAGES:
+-> WARNING at 2021-10-01 01:24:55.697880+00:00
 | output parser returned exit code<300>: The output model HDF5 file was not found.
```

Can anyone give some hints to solve this problem? 

Best regards,
Minkyu

**Comment by minkyu-p** [comment]
After many tries, I found a simple way to solve it. 

I deleted `.decode('utf-8')` from two files `~/.local/lib/python3.8/site-packages/fsc/hdf5_io/_save_load.py` and `~/.local/lib/python3.8/site-packages/fsc/hdf5_io/_subscribe.py`.
Then the workflow runs smoothly. 

I'm not sure whether this modification possibly leads to unexpected results for other cases.

**Comment by greschd** [comment]
Hi @minkyu-p. That sounds like an incompatibility between `h5py` (the HDF5 library) and `fsc.hdf5-io` (my own wrapper built on top of it).

There have been some breaking changes in `h5py`, and not all version combinations with `fsc.hdf5-io` are compatible. Which versions of the two packages do you have installed?

**Comment by minkyu-p** [comment]
@greschd Thank you for the reply.

I'm using `h5py` version 3.4.0 and `fsc-hdf5-io` version 1.0.1.

**Comment by greschd** [comment]
Interesting - there is an incompatibility between `h5py==3.4.0` and `fsc.hdf5-io==1.0.1` (fixed in `fsc.hdf5-io==1.0.2`), but the error is not what I would expect.

What seems to happen here is that `hdf5_handle[TYPE_TAG_KEY][()]` is already a `str`, not `bytes`. However, this should not be the case with `h5py` `3.0` or higher (see https://docs.h5py.org/en/stable/strings.html).

I have two follow-up questions:
- Were any of the input HDF5 files "old", i.e. maybe generated with another version? If yes, are they something you could share?
- Are you using the same Python install / virtualenv for AiiDA itself and for TBmodels? If not, are the versions of `h5py` and `fsc.hdf5-io` different?

**Comment by minkyu-p** [comment]
> Were any of the input HDF5 files "old", i.e. maybe generated with another version? If yes, are they something you could share?

I used the files in the example folder `examples/InSb` as it is. 

> Are you using the same Python install / virtualenv for AiiDA itself and for TBmodels? If not, are the versions of `h5py` and `fsc.hdf5-io` different?

I installed TBmodels only in my virtualenv for AiiDA. I also set `unset PYTHONPATH` for prepend text as you suggested in the document.

**Comment by greschd** [comment]
Unfortunately, I still haven't managed to reproduce the issue with this version combination.

However, I think there's no harm in performing the `.decode` operation only if it's not already a `str`. Can you try installing [this branch](https://github.com/FrescolinoGroup/pyhdf5io/tree/decode_only_if_needed) and see if it fixes things? If yes, I'll release it as `1.0.3`.

**Comment by minkyu-p** [comment]
@greschd Indeed, the fix works well. Thanks a lot for your effort!

**Comment by greschd** [comment]
Great, I released it as `1.0.3`: https://pypi.org/project/fsc.hdf5-io/1.0.3/
